crates/goose-cli/src/commands/acp.rs

```diff
@@ -694,45 +699,40 @@ impl acp::Agent for GooseAcpAgent {
 
         use futures::StreamExt;
 
-        // Track if we were cancelled
         let mut was_cancelled = false;
 
-        // Process the agent's response stream
         while let Some(event) = stream.next().await {
-            // Check if we've been cancelled
             if cancel_token.is_cancelled() {
                 was_cancelled = true;
                 break;
             }
 
             match event {
                 Ok(goose::agents::AgentEvent::Message(message)) => {
-                    // Re-acquire the lock to add message to conversation
+                    if let Err(e) = SessionManager::add_message(&session_id, &message).await {
+                        error!("Failed to write message to session: {}", e);
+                    }
```

Comment:
I don't think you want to do this. agent already keeps the conversation up to date.

Reply:
Updated